timesketch/lib/analyzers/geoip.py

```diff
@@ -0,0 +1,324 @@
+
+        return_fields = self.IP_FIELDS.copy()
+
+        events = self.event_stream(
+            query_string=query,
+            return_fields=return_fields
+        )
+
+        ip_addresses = defaultdict(lambda: defaultdict(list))
+
+        for event in events:
+            for ip_address_field in self.IP_FIELDS:
+                ip_address = event.source.get(ip_address_field)
+                if ip_address is None:
+                    continue
+
+                if isinstance(ip_address, str):
+                    ip_address = [ip_address]
+
+                for ip_addr in ip_address:
+                    if not self._validate_ip(ip_addr):
+                        logger.debug(
+                            f'Value {ip_addr} in {ip_address_field} not valid.')
+                        continue
+                    ip_addresses[ip_addr][ip_address_field].append(event)
+
+        logger.info(f'Found {len(ip_addresses)} IP address(es).')
+
+        for ip_address, ip_address_fields in ip_addresses.items():
+            with self.GEOIP_CLIENT() as client:
+                response = client.ip2geo(ip_address)
+
+            try:
+                iso_code, latitude, longitude, country_name, city_name = \
+                    response
+            except ValueError as error:
+                logging.error(f'GeoIP client must return 5 fields: '
+                    f'<iso_code, latitude, longitude, country_name, city_name>.'
+                    f' Number of fields returned: {len(response)}')
+            if response is None:
+                continue
+
+            if len(response) != 5:
+                logging.error('Invalid response from GeoIP client')
+
+            flag_emoji = emojis.get_emoji(f"FLAG_{iso_code}")
+
+            if flag_emoji is None:
+                logger.error(
+                    f'Invalid ISO code {iso_code} encountered for IP ' +
+                    f'{ip_address}.')
+
+            for ip_address_field, events in ip_address_fields.items():
+                for event in events:
+
+                    new_attributes = {}
+                    if latitude and longitude:
+                        new_attributes[
+                            f'{ip_address_field}_latitude'] = latitude
+                        new_attributes[
+                            f'{ip_address_field}_longitude'] = longitude
+                    if iso_code:
+                        new_attributes[
+                            f'{ip_address_field}_iso_code'] = iso_code
+                    if city_name:
+                        new_attributes[
+                            f'{ip_address_field}_city'] = city_name
+                    event.add_attributes(new_attributes)
+
+                    if flag_emoji:
+                        event.add_emojis([flag_emoji])
+
+                    if country_name:
+                        event.add_tags([country_name])
+                    event.commit()
+
+        return 'GeoIP analyzer completed.'
```

Comment:
Instead of using the logger above, just return the string here and it will be picked up in the UI as the result.

```suggestion
        return f'GeoIP analyzer completed: Found {len(ip_addresses)} IP address(es).'
```